File: 프로그래머스/0/181835. 조건에 맞게 수열 변환하기 3/조건에 맞게 수열 변환하기 3.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, int k) {
    vector<int> answer;
    //
    for (const auto& e : arr)
    {
        if (k % 2) answer.push_back(e * k);
        else answer.push_back(e + k);
    }
    //
    return answer;
}

File: 프로그래머스/0/181835. 조건에 맞게 수열 변환하기 3/조건에 맞게 수열 변환하기 3.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, int k) {
        ArrayList<Integer> list = new ArrayList<>();
        if (k % 2 == 0) {
            for (int i = 0; i < arr.length; i++) {
                list.add(arr[i] + k);
            }
        }  else {
            for (int i = 0; i < arr.length; i++) {
                list.add(arr[i] * k);
            }
        }
        int[] answer = new int[list.size()];
        for (int i = 0; i < answer.length; i++) {
            answer[i] = list.get(i);
        }
        return answer;
    }
}